src/JsVlcPlayer.cpp

```diff
@@ -840,7 +811,11 @@ void JsVlcPlayer::updateCurrentTime() {
 
 void JsVlcPlayer::setCurrentTime( libvlc_time_t time )
 {
-    _currentTime = std::max( 0ll, std::min( time, player().playback().get_length() ) );
+    const libvlc_time_t videoLength = player().playback().get_length();
+    if( 0 != videoLength )
+        _currentTime = std::max(0ll, std::min(time, videoLength));
+    else
+        _currentTime = time;
```

Comment:
_Problem:_ not rendering proper video frame when selecting the first clip
_Reason:_ `player().playback().get_length()` was 0 since video was not loaded yet.
_Solution:_ `clamp` the value only if `length` is available.